spec.md

```diff
@@ -177,6 +178,20 @@ circuit MyTop :
       ; ...
 ```
 
+## Externally Defined Circuits
+
+An externally defined circuit is a circuit declaration whose implementation
+exists elsewhere. It should contain only module declarations
+([@sec:externally-defined-modules] or [@sec:public-modules]).
+
+```firrtl
+extcircuit MyExtCircuit :
+  extmodule FooExtModule :
+      ; ...
+  extmodule BarExtModule :
+      ; ...
+```
```

Comment:
It would be great to also update the `include/firrtl.xml` grammar definition file so that syntax highlighting continues to work. This can be handled in a follow-up PR.

Reply:
I can try to do that in this PR! it seems straightforward